drivers/rf-transceiver/hmc630x/hmc630x.c

```diff
@@ -0,0 +1,711 @@
+int hmc630x_enable_temp(struct hmc630x_dev *dev, bool enable)
+{
+	int ret;
+	uint16_t tempflash, tempsensor;
+
+	switch (dev->type) {
+	case HMC6301:
+		tempflash = HMC6301_EN_TEMPFLASH;
+		tempsensor = HMC6301_ENBAR_TEMPS;
+		break;
+	default:
+	case HMC6300:
+		tempflash = HMC6300_EN_TEMPFLASH;
+		tempsensor = HMC6300_TEMPSENSOR_PWRDN;
+		break;
+	};
+
+	ret = hmc630x_write(dev, tempflash, enable);
+	if (ret)
+		return ret;
+
+	return hmc630x_write(dev, tempsensor, !enable);
+}
+
+/* Get a temperature reading. */
+int hmc630x_get_temp(struct hmc630x_dev *dev, uint8_t *temp)
+{
+	return hmc630x_read(dev, HMC630X_TEMPS, temp);
+}
+
+/* Enable FSK/MSK modulation inputs. */
+int hmc6300_enable_fm(struct hmc630x_dev *dev, bool enable)
+{
+	if (!dev)
+		return -EINVAL;
+
+	// this is a transmitter specific feature
+	if (dev->type != HMC6300)
+		return -ENOSYS;
+
+	return hmc630x_write(dev, HMC6300_ENABLE_FM, (uint8_t)enable);
+}
+
+/* Power On/Off the chip. */
+int hmc630x_enable(struct hmc630x_dev *dev, bool enable)
+{
+	int ret = -EFAULT;
+	uint8_t row;
+
+	if (dev->type == HMC6300) {
+		/* row4 */
+		row = no_os_field_prep(HMC630X_MASK(HMC6300_DRIVER_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6300_UPMIXER_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6300_IFVGA_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6300_DIVIDER_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6300_PA_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6300_RFVGA_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6300_TRIPLER_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6300_IF_UPMIXER_PWRDN), !enable);
+		ret = hmc630x_write_row(dev, HMC630X_ROW(HMC6300_DRIVER_PWRDN), row);
+		if (ret)
+			return ret;
+
+		ret = hmc630x_write(dev, HMC630X_EN_SYNTH_LDO, enable);
+	} else {
+		/* row0 */
+		row = no_os_field_prep(HMC630X_MASK(HMC6301_IFVGA_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6301_MIXER_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6301_BBAMP_PWRDN_I), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6301_TRIPLER_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6301_LNA_PWRDWN), !enable);
+		ret = hmc630x_write_row(dev, HMC630X_ROW(HMC6301_LNA_PWRDWN), row);
+		if (ret)
+			return ret;
+
+		/* row1 */
+		row = no_os_field_prep(HMC630X_MASK(HMC6301_BBAMP_SELL_ASK), enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6301_IFMIX_PWRDN_Q), enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6301_ASK_PWRDN), !enable);
+		row |= no_os_field_prep(HMC630X_MASK(HMC6301_IF_BGMUX_PWRDN), !enable);
+		ret = hmc630x_write_row(dev, HMC630X_ROW(HMC6301_BBAMP_SELL_ASK), row);
+	}
+
+	return ret;
+}
+
+/* Set the IF attenuation in steps from 0 (highest gain) to 15.
+ * For digital setting to dB correlation, see the datasheet.
+ */
+int hmc630x_set_if_attn(struct hmc630x_dev *dev, uint8_t attn)
+{
+	uint16_t param;
```

Comment:
nit: i'd separate variable declaration from the rest of the code with a new line

Reply:
fixed